torch/distributed/elastic/agent/server/local_elastic_agent.py

```diff
@@ -254,6 +263,17 @@ def _start_workers(self, worker_group: WorkerGroup) -> Dict[int, Any]:
             if "OMP_NUM_THREADS" in os.environ:
                 worker_env["OMP_NUM_THREADS"] = os.environ["OMP_NUM_THREADS"]
 
+            if "TORCHELASTIC_LOG_LINE_HEADER" in os.environ:
+                log_line_header = os.environ["TORCHELASTIC_LOG_LINE_HEADER"]
+
+                log_line_header = Template(log_line_header).safe_substitute(
+                    role_name=spec.role,
+                    rank=worker.global_rank,
+                    local_rank=local_rank,)
+                log_line_headers[local_rank] = log_line_header
+            else:
+                log_line_headers[local_rank] = f"[{spec.role}:{local_rank}]"
```

Comment:
shouldn't this be `f"[{spec.role}{local_rank}]:"` to be BC? (see my comment about moving the substitution into `TailLog`)